design-guidelines/interactive-element-states.md

```diff
@@ -0,0 +1,67 @@
+# Interactive Element States
+
+We can make our apps more intuitive by communicating the status of an interactive element, and providing feedback.
+
+## Common interaction subjects
+
+- Actions
+  - For example: buttons
+- Selectable items
+  - For example: items in a list of selectable options
+- Data inputs
+  - For example: a text input
+
+## States
+
+### Disabled
+
+Use a disabled state when an element cannot be selected or interacted with, but there is something that can be done to enable the element.
+
+It should be obvious that something is disabled before a user tries to interact with it.
+
+Disabled elements should not provide any feedback when they're clicked.
+
+Disabled elements should not be keyboard-navigable unless a tooltip is displayed when the disabled element is focus.
+
+#### When to disable an element
+
+An interactive element should be disabled when:
+
+- there is something a user can do in their current session to enable the element. For exapmle: disabling a button to add items to a list when a user has reached the maximum number of items. The button would be enabled after one or more existing items are removed.
+- something in progress needs to be completed before the element can be enabled. For example: disabling a "Save" button until the backend has sent an error or success response.
+- hiding the element might confuse users, or the user might have a strong expectation to execute that action. For example: if we have a list of items with dropdown menus, it could be confusing to hide a dropdown menu option in some of the items, but show it in others.
```

Comment:
Let's add that "Standard CRUD actions like View, Edit, Delete should be hidden if user does not have permission to access as there is should be no expectation to user that they should be able to access such actions"

Reply:
I'm adding your suggestion in the "When to hide an element" section